lib/src/main/java/com/tbruyelle/rxpermissions/ShadowActivity.java

```diff
@@ -1,13 +1,18 @@
 package com.tbruyelle.rxpermissions;
 
-import android.annotation.TargetApi;
-import android.app.Activity;
 import android.content.Intent;
-import android.os.Build;
+import android.content.pm.PackageManager;
 import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
 
-@TargetApi(Build.VERSION_CODES.M)
-public class ShadowActivity extends Activity {
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class ShadowActivity extends AppCompatActivity {
```

Comment:
What is the benefit for a non-ui activity to subclass `AppCompatActivity` ? I don't find any...